src/main/java/com/taskadapter/redmineapi/AttachmentManager.java

```diff
@@ -110,7 +110,41 @@ public Attachment uploadAttachment(String contentType, File content)
         }
     }
 
+/**
+	 * Uploads an attachment.
+	 * 
+	 * @param fileName
+	 *            file name of the attachment.
+	 * @param contentType
+	 *            content type of the attachment.
+	 * @param content
+	 *            attachment content stream.
+	 * @return attachment content.
+	 * @throws RedmineException
+	 *             if something goes wrong.
+	 * @throws IOException
+	 *             if input cannot be read. This exception cannot be thrown yet (I am not sure if http client can distinguish "network" errors and local errors) but is will be good to distinguish
+	 *             reading errors and transport errors.
```

Comment:
Please, explain me how this method could throw IOException and not RedmineException.

You promise to do that in javadoc. And this is another valid scenario. I (as a library user) want to show different messages to users of my application. It is one case when I could not upload file due to network problems (as network is not always reacheable on the mobile device). I could even provide automatic retries for this kind of errors. However, I want to provide a different error message if source file could not be read. It usually mean a problem with permanent storage (memory card, hard disk). This is a serious condition even for my app user. Or maybe user have no access to the file, there are no reasons to retry uploads (remember, I have automatic retries for "no-network" condition). Or maybe user ocasionally deleted the file between retries and there are no more reasons to try file upload any more.

This set of scenarios was supported by the previous method. These are functional requirements. What are reasons to drop this functional requirement? Is there another fix which do not introduce loss of functionality?